scipy/signal/_spectral_py.py

```diff
@@ -240,81 +244,106 @@ def lombscargle(
     # weight vector must sum to 1
     weights = weights / weights.sum()
 
-    # pre-allocate arrays for intermediate and final calculations
-    coswt = np.empty_like(x)
-    sinwt = np.empty_like(x)
-    pgram = np.empty_like(freqs)
-    # store a and b so that phase can be calculated outside loop, if necessary
-    a = np.empty_like(freqs)
-    b = np.empty_like(freqs)
-
     # if requested, perform precenter
     if precenter:
         y -= y.mean()
 
-    # calculate the single sum that does not depend on the frequency
-    Y_sum = (weights * y).sum()
+    # transform arrays
+    # row vector
+    freqs = freqs[np.newaxis, :]
+    # column vectors
+    x = x[:, np.newaxis]
+    y = y[:, np.newaxis]
+    weights = weights[:, np.newaxis]  # type: ignore
```

Comment:
Can do.

Reply:
Done.

Reply:
For some reason mypy doesn't catch that weights, at this point, is always as numpy array. It complains that `None` can't be indexed. I'll try again to see what happens, since I'm now calling reshape.